csharp/src/SeedLang/Interpreter/Compiler.cs

```diff
@@ -21,11 +21,21 @@
 namespace SeedLang.Interpreter {
   // The compiler to convert an AST tree to bytecode.
   internal class Compiler : AstWalker {
+    // The class to collect comparison information during compilation. The information includes
+    // operators and registers to hold the operand values, which is used to generate comparison
+    // notifications.
+    private class ComparisonInfo {
+      public uint FirstId;
+      public List<ComparisonOperator> Ops = new List<ComparisonOperator>();
+      public List<uint> ValueIds = new List<uint>();
+    }
```

Comment:
As we discussed before, the most tricky problem for a sorting algorithm visualizer about the comparison event is:

For a sorting algorithm visualizer, it's way useful to know what variables are being compared, instead of what evaluated values are being compared.

For example, as for the comparison `if a[i] > a[j]:` which is very common in compare-and-swap sorting algorithms, the sorting visualizer doesn't care much about the actual value of `a[i]` and `a[j]`. Instead, the visualizer cares very much about the semantic meaning of `a[i]` and `a[j]` themself - the `i`th element and the `j`th element of the data list `a`.

If it's not easy to pass semantic info out to the visualizer via notification event, it's still valid to use `V-Tag` to express additional info for a comparison event, such as:

```
# [[ Compare(a, i, j) ]]
if a[i] > a[j]:
  ...
```

Reply:
@wixette there are multiple cases for comparison expressions, for example:

```python
if 1 > 2:
if a > b:
if a + 1 > 100 / 3:
if a[1] < c:
if a[i] * 5 != b[j] * 6:
if a == True:
if a[i] < a[j]:
if i in (1, 2, 3):
```

It's hard to know which semantic info need for the visualizers. Maybe V-Tag can be used here.
And we need decide do we need support this kind of comparison notification or we only need support V-Tag for comparison expressions.